private/scripts/extras/problem_difficulty.py

```diff
@@ -0,0 +1,97 @@
+"""
+    Copyright (c) 2015-2020 Raj Patel(raj454raj@gmail.com), StopStalk
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+    THE SOFTWARE.
+"""
+
+from gluon import current
+from collections import defaultdict
+import time
+
+db = current.db
+
+pdtable = db.problem_difficulty
+ptable = db.problem
+sttable = db.suggested_tags
+ttable = db.tag
+
+codeforces_difficulty = {
+    "A": 1,
+    "B": 2,
+    "C": 3,
+    "D": 4,
+    "E": 5
+}
+
+tag_difficulty = {
+    "Ad-hoc": 1,
+    "Easy": 2,
+    "Medium": 3,
+    "Hard": 4.5
+}
+
+tags = tag_difficulty.keys()
+
+difficulty = defaultdict(list)
+
+# Compute problem difficulty for CodeForces problems.
+query = (ptable.link.like("%codeforces%")) & \
+        ~(ptable.link.like("%gymProblem%"))
+rows = db(query).select(ptable.id, ptable.link)
+
+for row in rows:
+    tag = row.link.split("/")[-1][0]
```

Comment:
`row.link.split("/")[-1]` this should give only one character string right? Don't need `[0]`

Reply:
Some CodeForces problems have A1, A2 and so on at the end of the URL, instead of just A. The [0] is to handle all those problems as well.